File: 2021/01/01.py
import sys

inp = list(map( int, (sys.stdin.readlines())))

part_1 = 0
depth = None

for line in inp:
    if depth != None and line > depth:
        part_1 += 1

    depth = line

print(f"Part 1: {part_1}")

part_2 = 0
depth = None

for i in range(len(inp) - 2):

    new_depth = sum(inp[i:i+3])

    if depth != None and new_depth > depth:
        part_2 += 1

    depth = new_depth

print(f"Part 2: {part_2}")

File: 2021/01/01.ts
const file = Bun.file("./01.in");

const depths = (await file.text()).split("\n").map(Number);

let part1 = 0;
let depth = undefined;

for (const line of depths) {
  if (depth && line > depth) {
    part1 += 1;
  }

  depth = line;
}

console.log(part1);
